lms/djangoapps/teams/models.py

```diff
@@ -0,0 +1,98 @@
+"""
+Django models related to teams functionality.
+"""
+
+from django.contrib.auth.models import User
+from django.db import models
+from django.utils.translation import ugettext_lazy
+from django.conf import settings
+from django_countries.fields import CountryField
+from .utils import slugify
+
+from xmodule_django.models import CourseKeyField
+
+
+class CourseTeam(models.Model):
+    """
+    This model represents team related info.
+    """
+
+    team_id = models.CharField(max_length=255, unique=True)
+    name = models.CharField(max_length=255)
+    is_active = models.BooleanField(default=True)
+    course_id = CourseKeyField(max_length=255, db_index=True)
+    topic_id = models.CharField(max_length=255, db_index=True, blank=True)
+    date_created = models.DateTimeField(auto_now_add=True)
+    # last_activity is computed through a query
+    description = models.CharField(max_length=1000)
+    country = CountryField(blank=True)
+    language = models.CharField(
+        max_length=16,
+        blank=True,
+        choices=settings.ALL_LANGUAGES,
+        help_text=ugettext_lazy("Optional language the team uses as ISO 639-1 code.")
+    )
+    users = models.ManyToManyField(User, db_index=True, related_name='teams', through='CourseTeamMembership')
+
+    @classmethod
+    def create(cls, name, course_id, description, topic_id=None, country=None, language=None):
+        """
+        Create a complete CourseTeam object.
+
+        Args:
+            name: Name of the team to be created
+            course_id: Course id
+            description: Description of the team
+            topic_id: Optional identifier for the topic the team formed around
+            country: Optional country where the team is based as ISO 3166-1 code
+            language: Optional language the team uses as ISO 639-1 code
+        """
+
+        team_id = slugify(name)
+        conflicts = cls.objects.filter(team_id__startswith=team_id).values_list('team_id', flat=True)
+
+        if conflicts and team_id in conflicts:
```

Comment:
Can this unique id generation be factored out into a helper method? It seems like useful functionality that we must need in other models.